jbpm-recommendation-spark-random-forest/src/main/java/org/jbpm/prediction/randomforest/SparkRandomForest.java

```diff
@@ -0,0 +1,213 @@
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.jbpm.prediction.randomforest;
+
+import com.google.common.collect.ImmutableList;
+import org.apache.spark.api.java.JavaSparkContext;
+import org.apache.spark.ml.Pipeline;
+import org.apache.spark.ml.PipelineModel;
+import org.apache.spark.ml.PipelineStage;
+import org.apache.spark.ml.classification.RandomForestClassifier;
+import org.apache.spark.ml.feature.OneHotEncoder;
+import org.apache.spark.ml.feature.StringIndexer;
+import org.apache.spark.ml.feature.VectorAssembler;
+import org.apache.spark.sql.Dataset;
+import org.apache.spark.sql.Row;
+import org.apache.spark.sql.RowFactory;
+import org.apache.spark.sql.SparkSession;
+import org.apache.spark.sql.types.DataTypes;
+import org.apache.spark.sql.types.StructField;
+import org.apache.spark.sql.types.StructType;
+import org.kie.api.task.model.Task;
+import org.kie.internal.task.api.prediction.PredictionOutcome;
+import org.kie.internal.task.api.prediction.PredictionService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class SparkRandomForest implements PredictionService {
+
+    public static final String IDENTIFIER = "SparkRandomForest";
+    private static final Logger logger = LoggerFactory.getLogger(SparkRandomForest.class);
+
+    private final SparkSession session;
+
+    private final StructType trainingSchema;
+    private final StructType predictionSchema;
+    private org.apache.spark.sql.Dataset<Row> df;
+    private int count = 0;
+    private PipelineModel pm = null;
+    private final Pipeline pipeline;
+    private double confidenceThreshold = 1.0;
+    private final int numTrees = 10;
+    private final int MINIMUM_OBSERVATIONS = 2;
+
+
+    public SparkRandomForest() {
+
+        final String sparkMaster = System.getProperty("org.jbpm.prediction.randomforest.spark.master", "local[4]");
+
+        logger.info("Starting Apache Spark at {}", sparkMaster);
+
+        session = SparkSession.builder()
+                .master(sparkMaster)
+                .appName("Random Forest example")
+                .getOrCreate();
+
+        // Get the Java specific Spark context
+        final JavaSparkContext jsc = new JavaSparkContext(session.sparkContext());
+
+        // Dataframe schema
+        trainingSchema = DataTypes.createStructType(
+                new StructField[] {
+                        DataTypes.createStructField("user", DataTypes.StringType, false),
+                        DataTypes.createStructField("level", DataTypes.IntegerType, false),
+                        DataTypes.createStructField("label", DataTypes.IntegerType, false),
```

Comment:
Why this structure doesn't contain "item"? If "item" is not used for training the model, perhaps it may be removed from the readme and tests for clarity.